docs/Tutorial_DAMP.ipynb

```diff
@@ -0,0 +1,778 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #96.    def _backward_processing(
</pre><p>The flow of this code feels weird and non-obvious to me what is happening. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/872/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
<p>I will change the code back to <code>while</code> (original format) and see if I can find other ways to simplity the code. We can then start from there and enhance it if needed. </p>

Reply:
I kept the for-loop but revised the code to make it more readable. Let's see how you feel about it.